helix-term/src/commands.rs

```diff
@@ -167,6 +167,7 @@ impl MappableCommand {
             Self::Typable { name, args, doc: _ } => {
                 let args: Vec<Cow<str>> = args.iter().map(Cow::from).collect();
                 if let Some(command) = typed::TYPABLE_COMMAND_MAP.get(name.as_str()) {
+                    commit_undo_checkpoint(cx);
```

Comment:
Should only be called if in insert mode, together with a comment explaining why this is needed in the first place